core/src/main/java/de/bwaldvogel/mongo/backend/AbstractMongoDatabase.java

```diff
@@ -587,7 +587,7 @@ private Document commandChangeStreamPipeline(Document query, Oplog oplog, String
         Document cursorDocument = (Document) query.get("cursor");
         int batchSize = (int) cursorDocument.getOrDefault("batchSize", 0);
 
-        Cursor cursor = oplog.createCursor(changeStreamDocument);
+        Cursor cursor = oplog.createCursor(changeStreamDocument, String.format("%s.%s", databaseName, collectionName));
```

Comment:
Could you deduplicate the namespace string with the line below? (`getDatabaseName() + "." + collectionName`)